xbmc/interfaces/json-rpc/schema/methods.json

```
@@ -2342,6 +2342,18 @@
     ],
     "returns": "string"
   },
+  "Input.ButtonEvent": {
+    "type": "method",
+    "description": "Send a button press event",
+    "transport": "Response",
+    "permission": "Navigate",
+    "params": [
+      { "name": "button", "type": "string", "required": true, "description": "Button name" },
+      { "name": "keymap", "type": "string", "required": true, "description": "Keymap name (KB, XG, R1, or R2)" },
```

Is the list of possible "keymap" values static enough that it could be declared with an `enum`?
```suggestion
      { "name": "keymap", "type": "string", "required": true, "description": "Keymap name (KB, XG, R1, or R2)", "enum": [ "KB", "XG", "R1", "R2" ] },
```

---

I think supporting the universal remote map makes this not feasible, since that one takes button strings of the form "obc<number>".

For the others it should be pretty static, but the list is quite long (probably 150-200 in total)

---

Ah, sorry, I thought you meant the buttons rather than the keymap.  The list of keymaps is indeed static and can be an enum as you suggest.